docs/internals/variable_cleanup.rst

```diff
@@ -28,25 +29,45 @@ the boolean values before they are used as the condition for
-|Type           |Valid Values   |Invalid Values Mean|
-+===============+===============+===================+
-|enum of n      |0 until n - 1  |exception          |
-|members        |               |                   |
-+---------------+---------------+-------------------+
-|bool           |0 or 1         |1                  |
-+---------------+---------------+-------------------+
-|signed integers|sign-extended  |currently silently |
-|               |word           |wraps; in the      |
-|               |               |future exceptions  |
-|               |               |will be thrown     |
-|               |               |                   |
-|               |               |                   |
-+---------------+---------------+-------------------+
-|unsigned       |higher bits    |currently silently |
-|integers       |zeroed         |wraps; in the      |
-|               |               |future exceptions  |
-|               |               |will be thrown     |
-+---------------+---------------+-------------------+
+The following table describes the cleaning rules applied to different types,
+where ``higher bits`` refers to the remaining bits in case the type has less than 256 bits.
+
++---------------+---------------+-------------------------+
+|Type           |Valid Values   |Cleanup of Invalid Values|
++===============+===============+=========================+
+|enum of n      |0 until n - 1  |throws exception         |
+|members        |               |                         |
++---------------+---------------+-------------------------+
+|bool           |0 or 1         |results in 1             |
++---------------+---------------+-------------------------+
+|signed integers|higher bits    |currently silently       |
+|               |set to the     |signextends to a valid   |
+|               |sign bit       |value, i.e. all higher   |
+|               |               |bits are set to the sign |
+|               |               |bit; may throw an        |
+|               |               |exception in the future  |
++---------------+---------------+-------------------------+
+|unsigned       |higher bits    |currently silently masks |
+|integers       |zeroed         |to a valid value, i.e.   |
+|               |               |all higher bits are set  |
+|               |               |to zero; may throw an    |
+|               |               |exception in the future  |
++---------------+---------------+-------------------------+
+
+Note that valid and invalid values are dependent on their type size.
+Consider ``uint8``, the unsigned 8-bit type, which has the following valid values:
+
+.. code-block:: none
+
+    0000...0000 0000 0000
+    0000...0000 0000 0001
+    0000...0000 0000 0010
+    ....
+    0000...0000 1111 1111
+
+Any invalid value will have the higher bits set to zero:
+
+.. code-block:: none
+
+    0101...1101 0000 0000   invalid value
+    0000...0000 0000 0000   cleaned value
```

Comment:
Maybe add signed integer examples to clarify the "signextend" term?

```suggestion
    0000...0000 0000 0000   cleaned value
    
For positive 8-bit signed integers:
.. code-block:: none
    Valid values:
    0000...0000 0000 0001
    0000...0000 0000 0010
    ....
    0000...0000 0111 1111

    invalid value: 0101...1101 0000 0100
    sign bit: 0
    cleaned value: 0000...0000 0000 0100

For negative 8-bit signed integers:
.. code-block:: none
    Valid values:
    1111...1111 1111 1111
    1111...1111 1111 1110
    ....
    1111...1111 1000 0000

    invalid value: 0000...0000 1111 1111
    sign bit: 1
    cleaned value: 1111...1111 1111 1111
```

Reply:
Done.